Tests/WhisperKitTests/RegressionTests.swift

```diff
@@ -0,0 +1,199 @@
+                let timeElapsed = Date().timeIntervalSince(startTime)
+                memoryStats.measure(from: currentMemoryValues, timeElapsed: timeElapsed)
+                latencyStats.measure(from: currentTPSValues, timeElapsed: timeElapsed)
+                currentMemoryValues = []
+                currentTPSValues = []
+            }
+            return true
+        }
+        
+        let whisperKit = try await WhisperKit(model: model)
+        memoryStats.preTranscribeMemory = Float(systemMemoryChecker.getMemoryUsed())
+        
+        let transcriptionResult = try await XCTUnwrapAsync(
+            await whisperKit.transcribe(audioPath: audioFilePath, callback: callback),
+            "Transcription failed"
+        )
+        XCTAssert(transcriptionResult.text.isEmpty == false, "Transcription failed")
+        
+        memoryStats.postTranscribeMemory = Float(systemMemoryChecker.getMemoryUsed())
+        let testInfo = TestInfo(
+            device: WhisperKit.deviceName(),
+            audioFile: audioFilePath,
+            model: model,
+            date: startTime.formatted(Date.ISO8601FormatStyle().dateSeparator(.dash)),
+            timeElapsedInSeconds: Date().timeIntervalSince(startTime),
+            timings: transcriptionResult.timings
+        )
+        let json = RegressionStats(testInfo: testInfo, memoryStats: memoryStats, latencyStats: latencyStats)
+        do{
+            let attachment = try XCTAttachment(data: json.jsonData())
+            attachment.lifetime = .keepAlways
+            add(attachment)
+        }
+        catch{
+            XCTFail("Failed with error: \(error)")
+        }
+    }
+    
+    //MARK: Distil Whisper
+    func testDistilLargeV3PerformanceOverTime() async throws{
+        try await testAndMeasureModelPerformance(model: "distil-whisper_distil-large-v3")
+    }
+    
+    func testDistilLargeV3_594MBPerformanceOverTime() async throws{
+        try await testAndMeasureModelPerformance(model: "distil-whisper_distil-large-v3_594MB")
+    }
+    
+    func testDistilLargeV3_TurboPerformanceOverTime() async throws{
+        try await testAndMeasureModelPerformance(model: "distil-whisper_distil-large-v3_turbo")
+    }
+    
+    func testDistilLargeV3_Turbo_600MBPerformanceOverTime() async throws{
+        try await testAndMeasureModelPerformance(model: "distil-whisper_distil-large-v3_turbo_600MB")
+    }
+    
+    //MARK: Open AI Whisper
+    func testBasePerformanceOverTime() async throws{
+        try await testAndMeasureModelPerformance(model: "base")
+    }
+    
+    func testBaseEnglishPerformanceOverTime() async throws{
+        try await testAndMeasureModelPerformance(model: "base.en")
+    }
+    
+    func testLargeV2PerformanceOverTime() async throws{
+        try await testAndMeasureModelPerformance(model: "large-v2")
+    }
+    
+    func testLargeV2949PerformanceOverTime() async throws{
+        try await testAndMeasureModelPerformance(model: "large-v2_949")
```

Comment:
These are all subject to change over time unfortunately, so these tests should actually search the full HF repo for all models available in there in order to run these tests. You can use this method to find all available models: https://github.com/argmaxinc/WhisperKit/blob/8564ce2bbec03649526678f51f4e249e9e052a6c/Sources/WhisperKit/Core/WhisperKit.swift#L131-L136

Reply:
Updated the code to use this function, Zack